### DISCUSSION TITLE
debtCapModifier: 1.0|upkeepModifier: 1.0|townBlockBuyBonusLimit: 0| What is it?

### DISCUSSION THREAD

**Discussion by ProblemsSender** [TownyAdvanced/Towny#5232] [Q&A] [answered]
And why do both nation and town have that statement? Does nation affect the overall cost that a town member has to pay? What is the formula?
*Update: Does "townBlockBuyBonusLimit" have to be higher for each level of the town?

**Comment by LlmDl** [comment, marked as answer]
Nations and towns can both pay upkeep costs to the server to remain in existence.

Town upkeepModifier affects the town's cost, the nation upkeepModifier affects the nation's cost.

All of the town and nation_level strings are explained: https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works#configuring-town_level-and-nation_level

Most people would make the townBlockBuyBonusLimit increase for higher levels.

**Reply by ProblemsSender** [reply]
> Nations and towns can both pay upkeep costs to the server to remain in existence.
> 
> Town upkeepModifier affects the town's cost, the nation upkeepModifier affects the nation's cost.
> 
> All of the town and nation_level strings are explained: https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works#configuring-town_level-and-nation_level
> 
> Most people would make the townBlockBuyBonusLimit increase for higher levels.

About townBlockBuyBonusLimit, what would happen if I set
lv 1 : townBlockBuyBonusLimit: 1000
lv 2 : townBlockBuyBonusLimit: 2000

Does it cost 3000 for lv 1 and lv 2? Or just 2000 for both? (Because I have to overthink that after each upgrade means the cost is reserved(that's why it is called level, just like numResidents))

And for upkeepModifier, does it means that if I have a nation and having the settings like this
Town : upkeepModifier: 1.0
Nation : upkeepModifier: 2.0
I would have to pay 3.0 times the cost every day?

Thanks!

**Reply by LlmDl** [reply]
The town_level and nation_levels do not stack. The values are applied to a town or nation when that town or nation has the required number of residents to attain the town_level or nation_level's numResidents.

The townBlockBuyBonusLimit of 1000 means a town at that level can buy up to 1000 townblocks using /t buy bonus.

Town upkeepModifiers affect the town's upkeep.
Nation upkeepModifiers affect the nation's upkeep.

**Reply by ProblemsSender** [reply]
Ok I've got it LlmDl.
